lib/schedule.js

```diff
@@ -508,14 +523,14 @@ function cancelInvocation(invocation) {
 /* Recurrence scheduler */
 function scheduleNextRecurrence(rule, job, prevDate, endDate) {
 
-  prevDate = (prevDate instanceof Date) ? prevDate : (new Date());
+  prevDate = (prevDate instanceof CronDate) ? prevDate : (new CronDate(undefined, rule.timezone || null));
 
```

Comment:
What if `prevDate` is undefined and `rule` is not a `RecurrenceRule`? Would it work?